File: composeApp/src/androidMain/kotlin/org/gce/racehub/theme/Dimens.kt
package org.gce.racehub.theme

import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.ui.unit.dp

/**
 * Centralized spacing, radius and sizing tokens for the whole app.
 *
 * Screens must use these instead of hardcoding dp values so gutters, corner
 * radii and vertical rhythm stay consistent across every screen (and in sync
 * with the iOS `Theme.Spacing` / `Theme.Radius` constants).
 */
object Dimens {
    /** Standard horizontal screen gutter used by every screen. */
    val screenGutter = 20.dp

    // Vertical rhythm — use these for spacing between elements.
    val spaceXs = 4.dp
    val spaceSm = 8.dp
    val spaceMd = 12.dp
    val spaceLg = 16.dp
    val spaceXl = 24.dp

    // Corner radii.
    /** Cards / primary surfaces. */
    val radiusCard = 16.dp
    /** Inner tiles, stat boxes, segmented controls. */
    val radiusTile = 12.dp
    /** Small chips, badges, tags, buttons. */
    val radiusChip = 8.dp

    // Reusable shapes built from the radii above.
    val cardShape = RoundedCornerShape(radiusCard)
    val tileShape = RoundedCornerShape(radiusTile)
    val chipShape = RoundedCornerShape(radiusChip)
}

File: iosApp/iosApp/Theme/Dimens.swift
import CoreGraphics

/// Centralized spacing, radius and sizing tokens for the whole app.
///
/// Views must use these instead of hardcoding point values so gutters, corner
/// radii and vertical rhythm stay consistent across every screen (and in sync
/// with the Android `Dimens` object).
enum Theme {

    enum Spacing {
        /// Standard horizontal screen gutter used by every screen.
        static let screenGutter: CGFloat = 20

        // Vertical rhythm — use these for spacing between elements.
        static let xs: CGFloat = 4
        static let sm: CGFloat = 8
        static let md: CGFloat = 12
        static let lg: CGFloat = 16
        static let xl: CGFloat = 24
    }

    enum Radius {
        /// Cards / primary surfaces.
        static let card: CGFloat = 16
        /// Inner tiles, stat boxes, segmented controls.
        static let tile: CGFloat = 12
        /// Small chips, badges, tags, buttons.
        static let chip: CGFloat = 8
    }
}
